### ISSUE TITLE
[QUESTION] Can I replace a media message with a text message?

### ISSUE THREAD

**Issue by ingroxd** [python-telegram-bot/python-telegram-bot#2406]
### Issue I am facing
I am trying to edit a message through its `chat_id`+`message_id`, but I am prompted with a `BadRequest` error.
In particular, PTB yell at me that there is no text to edit.

Tecnically it is right, because of the orginal message is a media message: there are the media (a photo in my specific case) and the caption (with or without inline keyboard).
Any way if I try to go the other way around it does not complain.

To better explain myself, if I try to replace a simple text message (with or without a `reply_markup`) with a media message there is no problem at all.
If I try to replace a media message with a simple text message I get the aforementioned error.

I am currently using a workaround (delete + new send).
Is there something I am missing?

### Traceback to the issue
```
telegram.BadRequest: "There is no text in the message to edit"
```

### Related part of your code
```python
context.bot.edit_message_text(
    chat_id=chat_id,
    message_id=message_id,
    text=text,
    reply_markup=reply_markup
)
```

**Comment by starry69** [comment]
Iirc you can't edit media message to normal text message, as error says: `there is no text in msg to edit` since media message has `caption` not `text` and there is nothing to edit for `editMessageText` method.

**Comment by ingroxd** [comment]
Yup, I noticed; the error is crystal clear.
I was wondering if there is something I can do to avoid this type of error, like manipulate the media message with another media message adding (instead of editing) the text, or something like that.

Maybe something like:
- Delete photo (not the message)
- Add arbitrary useless text (like "qwe")
- Modify the text

I have this doubt because if I edit a text message with a media message containing a photo, the text is automatically trimmed and the image is added.

**Comment by starry69** [comment]
But questions is, can you really edit a text message to media message? Iirc it should raise `telegram.error.BadRequest: There is no media in the message to edit`. Going back to your first point, yes, you can surely edit/manipulate some media message to another media message.

**Comment by ingroxd** [comment]
> But questions is, can you really edit a text message to media message? Iirc it should raise `telegram.error.BadRequest: There is no media in the message to edit`.

I confirm that I can convert a text message into a media message using the following:
```python
context.bot.edit_message_media(
    chat_id=chat_id,
    message_id=message_id,
    media=InputMediaPhoto(self.get_photo())
)
```

I actually add a caption, too, in order to have text and inline buttons, like so:
```python
context.bot.edit_message_caption(
    chat_id=chat_id,
    message_id=message_id,
    caption=text,
    reply_markup=reply_markup
)
```

Due to the fact that I can actually convert a text message to a media message I was wondering if it is possible to do the same thing on the other way.

Last time I checked the Telegram Bot API I coudn't find the error `There is no text in the message to edit`, but it was a long time ago.
Is it something added by PTB or are actually the Bot API?

**Comment by Poolitzer** [comment]
*me, reading this message*: What, no. What? Noo. What???

```python

from telegram import Bot, InputMediaPhoto

bot = Bot("Bottoken")
msg = bot.send_message(208589966, "test")
msg.edit_media(media=InputMediaPhoto(media="https://github.com/python-telegram-bot/python-telegram-bot/blob/master/tests/data/game.png"))

```

gives me the expected `telegram.error.BadRequest: There is no media in the message to edit`.

And you are telling me there is a way to not get that error? This is mind blowing, something I have never ever seen before, please give me a working example.

**Comment by starry69** [comment]
> And you are telling me there is a way to not get that error? This is mind blowing, something I have never ever seen before, please give me a working example.

+1

**Comment by ingroxd** [comment]
> And you are telling me there is a way to not get that error? This is mind blowing, something I have never ever seen before, please give me a working example.

I have to retract...
I had wrong information (I based this issue on what the tester wrote me).
I actually tried it myself: I wrote a small piece of code only for this issue and I can confirm that the bot CANNOT modify the last message adding the photo sa previously stated, but send a new message instead (We are talking abbout text -> media).
The error is exaclty as you said:
```
BadRequest: There is no media in the message to edit
```

This is perfectly coherent to the fact the it is not possible to go the other way around (media -> text).

I am really sorry and I apologize for the inconvenience.

The question, anyway still remain: Is there any other workaround I could use instead of delete + send?
